src/pages/Loans/LoansOverview/hooks/use-loan-form-data.tsx

```diff
@@ -118,6 +119,7 @@ const useLoanFormData = (
     governanceBalance,
     transactionFee,
     assetPrice,
+    balance: loanAction === 'repay' ? (assetBalance.gt(maxAmount) ? maxAmount : assetBalance) : maxAmount,
```

Comment:
Could this be moved into `getMaxAmount` instead, so there would be no need to add `balance` field?

Also L122: `pickSmallerAmount` util can be used here to simplify the nested ternary :+1: